kairos-cli/src/crypto/private_key.rs

```diff
@@ -0,0 +1,13 @@
+use std::path::Path;
+
+use super::error::CryptoError;
+
+pub struct CasperPrivateKey(pub casper_types::SecretKey);
+
+impl CasperPrivateKey {
+    pub fn from_file<P: AsRef<Path>>(file_path: P) -> Result<Self, CryptoError> {
+        casper_types::SecretKey::from_file(file_path)
+            .map(Self)
+            .map_err(|error| error.into())
```

Comment:
Return `ErrorExt` not the more general `CryptoError`.
We want our errors to be as precise as possible within reason, such that if you match on it you don't have to handle impossible error cases.

Ideally a function's error enum should only contain variants that the function may construct.
This ideal is not always the reasonable, a frequent compromise is to write an error per module.
`ErrorExt` has variant's `casper_types::SecretKey::from_file` never constructs, but I would not suggest spending time fixing it.

Reply:
My intention was to isolate Casper related types in `crypto` module, which introduced intermediary `CryptoError`. Returning `ErrorExt` directly would make all 3 command handlers more complex, because error has to be upcasted manually:

```rs
let _signer = CasperSigner::from_file(args.private_key_path.field)
        .map_err(|error| CliError::CryptoError { error: error.into() })?;
```

We could avoid it by merging `CryptoError` with `Error`, but not sure if this is the thing you want.

Reply:
Yeah you would have to add `.map_err(CryptoError::from)` to each handler.

> If I update crypto related function to return ErrorExt directly, then I am polluting whole project with Casper specific type
If the goal is to encapsulate `casper_types` like `ErrorEx` to guard callers against future upstream changes then `ErrorExt` cannot appear in CryptoError at all. It may be nested, but anything in a enum is public.

The point of using a static error type (the thiserror style) is to be able to match on the errors and handle exactly what went wrong. An overly general error return does not tell the caller what might go wrong. This leaves the caller to look through the code to find out what each function might return; like I did with `ErrorExt`.

We may not want precise errors, maybe we're too early, or we don't expect anyone to use them. 
Given this is CLI app code not library code we could choose to use `anyhow` or another dynamic error type.
If we are going to have general errors I'd suggest that for future work.

For now I'd be okay with leaving the return as `CryptoError` and changing it when we need it.